test/test_dht.cpp

```diff
@@ -3687,6 +3687,7 @@ TORRENT_TEST(sample_infohashes)
 
 	g_sent_packets.clear();
 
+	node_id nid = t.dht_node.nid();
```

Comment:
this is *our* node ID. not the node ID that will respond to the sample-infohashes request

Reply:
The node we're sending to is `udp::endpoint initial_node = rand_udp_ep();`, correct? So how can we determine its id?